### PR TITLE
feat(gemma4): add E2B FFT recipe; make E4B FFT save disk-safe

### PR DESCRIPTION
**Pull request by aniruddh-alt**
# Description

Extends Gemma 4 full fine-tuning to the Efficient series:

- **New `e2b_full` recipe** (`configs/recipes/gemma4/sft/e2b_full/{train,gcp_job}.yaml`) — full fine-tune for `google/gemma-4-E2B-it` (FSDP `FULL_SHARD`, 4×A100), mirroring `e4b_full`.
- **`e4b_full` made disk-safe**: `save_steps 200 → 0` + `save_epoch false` + `save_final_model true`, so it writes only the final model instead of large intermediate FSDP optimizer checkpoints (matching the gemma2/gemma3 full recipes). Also dropped the stale gradient-accumulation loss-inflation note — `Gemma4ForConditionalGeneration` sets `accepts_loss_kwargs=False` as of transformers 5.6.0 (huggingface/transformers#40564; verified on-cluster), so every install satisfying oumi's pins (uv override resolves to 5.7.x, plain pip to 5.9.x) gets the fix. The recipe headers now state `transformers >= 5.6` with a one-line warning, since 5.5.x is the remaining affected line.
- README: E2B now lists **FFT + LoRA**; adds the E2B FFT launch commands.

Both are the standard multimodal gemma4 arch (`-it` checkpoints with their own chat template, FSDP `transformer_layer_cls: Gemma4TextDecoderLayer`). Hyperparameters follow the gemma FFT family (LR 1.5e-5, weight_decay 0.05, cosine + 0.05 warmup, `adamw_torch_fused`).

## Validation

Validated on the real `-it` checkpoints (8×H100 FSDP `FULL_SHARD`, FFT vs base, generation accuracy on 500 test examples, recipe hyperparameters, 3 epochs; banking77 full train split, PubMedQA 5k `pqa_artificial` evaluated on the disjoint `pqa_labeled`). FFT improves over the already-instruct-tuned base on every cell:

| Model | Task | Base | FFT | Δ |
|---|---|---|---|---|
| E2B | banking77 | 0.652 | **0.916** | **+0.264** |
| E2B | PubMedQA | 0.344 | **0.714** | **+0.370** |
| E4B | banking77 | 0.742 | **0.954** | **+0.212** |
| E4B | PubMedQA | 0.640 | **0.768** | **+0.128** |

Companion to the 12B FFT recipe (#2495).

## Before submitting

- [x] This PR only changes configuration + documentation (recipe YAML + README; no code, no tests needed).
- [x] Read the contributor guidelines.
- [x] Related work linked above.

> [!NOTE]
> **Liberate**
> Risk: low

### CODE REVIEW

File: configs/recipes/gemma4/sft/e4b_full/train.yaml [hunk 1]
```diff
@@ -34,13 +34,11 @@ data:
...
+  save_steps: 0
+  save_epoch: false
+  save_final_model: true
   num_train_epochs: 1
   per_device_train_batch_size: 1
-  # NOTE: gradient accumulation inflates reported loss for Gemma 4 (~4x) due
-  # to a missing `accepts_loss_kwargs = False` on Gemma4ForConditionalGeneration.
-  # Fixed on transformers main but not yet released as of 5.5.4. See
-  # huggingface/transformers#40564 (same bug existed in Gemma 3).
   gradient_accumulation_steps: 4
```
[the remark this answers is not in the document]
Comment (aniruddh-alt) [reply configs/recipes/gemma4/sft/e4b_full/train.yaml:43]: Partly right — checked the transformers source per release tag: `Gemma4ForConditionalGeneration.accepts_loss_kwargs = False` landed in **v5.6.0**, not 5.10 (the PR body had that wrong; fixed). So a fresh install under oumi's pins is fine: the uv override (`>=5.5,<5.8`) resolves to 5.7.x and plain pip (`<5.10`) to 5.9.x — both contain the fix. The real gap was that these recipe headers advertised `transformers >= 5.5.4`, the one line that still has the bug. Bumped both `e2b_full` and `e4b_full` headers to `>= 5.6` with a one-line warning in 5e3277773. No oumi pin bump needed.

**Review by wizeng23 — APPROVED** [review]

**Review by lefft — APPROVED** [review]
